ci_tutorial/tutorial.rst

```diff
@@ -0,0 +1,141 @@
+      This tutorial is not the unique way that things should be done.
+      Different organizations might have different needs and priorities, different build services power and budget, different sizes, etc.
+      The principles and practices presented in the tutorial might need to be adapted.
+    - However some of the principles and best practices would be general for all approaches. Things like package immutability, using promotions
+      between repositories and not using the ``channel`` for that purpose are good practices that should be followed.
+
+
+Packages and products pipelines
+-------------------------------
+
+When a developer is doing some changes to a package source code, we will consider 2 different parts or pipelines of the overall system CI:
+the **packages pipeline** and the **products pipeline**
+
+
+The **packages pipeline** will build, create and upload the package binaries for the different configurations and platforms, when some
+developer is submitting some changes to one of the organization repositories source code. For example if a developer is doing some changes
+to the ``ai`` package, improving some of the library functionality, and bumping the version to ``ai/1.1.0``. If the organization needs to
+support both Windows and Linux platforms, then the package pipeline will build the new ``ai/1.1.0`` both for Windows and Linux, before
+considering the changes are valid. If some of the configurations fail to build under a specific platform, it is common to consider the
+changes invalid and stop the processing of those changes, until the code is fixed.
+
+
+The **products pipeline** responds a more challenging question: does my "products" build correctly with the latest changes that have been done
+to the packages? This is the real "Continuous Integration" part, in which changes in different packages are really tested against the organization
+important product to check if things integrate cleanly or break. Let's continue with the example above, if we now have a new ``ai/1.1.0`` package,
+is it going to break the existing ``game/1.0`` and/or ``mapviewer/1.0`` applications? Is it necessary to re-build from source some of the existing
+packages that depend directly or indirectly on ``ai`` package? In this tutorial we will use ``game/1.0`` and ``mapviewer/1.0`` as our "products",
+but this concept will be further explained later, and specially why it is important to think in terms of "products" instead of trying to explicitly
+model the dependencies top-bottom in the CI.
+
+
+Repositories and promotions
+---------------------------
+
+The concept of multiple server side repositories is very important for CI. In this tutorial we will use 3 repositories:
+
+- ``develop``: This repository is the main one that developers have configured in their machines to be able to ``conan install`` dependencies
```

Comment:
I don't know if "develop" is a great name for this. Users might not necessarily associate develop with something that is supposed to be stable and the front facing remote for their daily usage